lib/entitlements/backend/github_team/service.rb

```diff
@@ -35,8 +35,9 @@ class TeamNotFound < RuntimeError; end
         def initialize(addr: nil, org:, token:, ou:, ignore_not_found: false)
           super
           Entitlements.cache[:github_team_members] ||= {}
-          Entitlements.cache[:github_team_members][org] ||= {}
-          @team_cache = Entitlements.cache[:github_team_members][org]
+          Entitlements.cache[:github_team_members][addr] ||= {}
+          Entitlements.cache[:github_team_members][addr][org] ||= {}
+          @team_cache = Entitlements.cache[:github_team_members][addr][org]
```

Comment:
🎉 Thank you for this!

There is a method on the base GitHub service class use use to handle this collision for org membership, let's use that instead for consistency.

```suggestion
          Entitlements.cache[:github_team_members][org_signature] ||= {}
          @team_cache = Entitlements.cache[:github_team_members][org_signature]
```